Add Clickhouse state store docs v1.16

middt opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

<!--Please explain the changes you've made-->

## Issue reference

<!--Please reference the issue this PR will close: #[issue number]--> https://github.com/dapr/components-contrib/pull/3675

msfussell commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
@middt - I presume this is the file that you want to add, along with the index chart. Can you remove all the other files that have been pulled in as part of this PR?

msfussell commented on daprdocs/data/components/state_stores/generic.yaml:
@middt - I presume this is the file that you want to add, along with the content file. Can you remove all the other files that have been pulled in as part of this PR?

middt commented:
@msfussell I've removed the unnecessary files—only the required ones remain. Thanks again for the feedback!

msfussell commented on daprdocs/data/components/state_stores/generic.yaml:
```diff
@@ -20,6 +20,17 @@
+  link: setup-clickhouse
+  state: Alpha
+  version: v1
+  since: "1.15"
```
This is first released in v1.16 yes?
```suggestion
  since: "1.16"
```

msfussell commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,204 @@
+    value: <USERNAME>
+  - name: password # Optional
+    value: <PASSWORD>
+  # Uncomment this if you wish to use ClickHouse as a state store for actors (optional)
```
You can remove all the comments on using actors, since this is not a transactional state store. 
```suggestion
```

msfussell commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,204 @@
+The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
+{{% /alert %}}
+
+If you wish to use ClickHouse as an actor store, append the following to the yaml.
```
Same here. You can remove all the comment/section on using actors, since this is not a transactional state store.

msfussell reviewed (commented):
@middt - Great docs. A few suggested edits. Appreciate the contribution

middt commented:
@msfussell Thank you for the great feedback! 🙏
I've addressed all your suggestions:
✅ Fixed version: Changed since: "1.15" to since: "1.16" in generic.yaml 
✅ Removed actor references: Removed all actor-related comments and sections since ClickHouse is not a transactional state store

Thanks for the thorough review and for helping make this contribution better!

msfussell reviewed (approved):
@middt - Appreciate this amazing contribution to Dapr! Would love to get this to be a stable component in due course.

middt commented:
> @middt - Appreciate this amazing contribution to Dapr! Would love to get this to be a stable component in due course.

@msfussell – Thank you so much for the approval and kind words! 🙌

msfussell commented:
@middt - can you remove the sdkdoc/python file from the PR. This should not be here, otherwise cannot merge

middt committed:
Add ClickHouse state store documentation

- Add comprehensive documentation for ClickHouse state store component
- Include setup instructions for self-hosted, Kubernetes, and cloud deployments
- Document supported features: CRUD, ETag, and TTL support
- Add advanced configuration options and performance considerations
- Update component registry with ClickHouse entry in generic.yaml
- Set correct version as 1.16 (first release)
- Remove actor-related content as ClickHouse is not transactional

Signed-off-by: Mehmet TOSUN <mehmet.tosun@gmail.com>

middt commented:
> msfussell

@msfussell ✅ Done! Removed sdkdocs/python and all other unnecessary files. The PR now contains only the 2 ClickHouse files. Thanks for catching that!

middt committed:
Merge branch 'v1.16' into add-clickhouse-state-store-docs-v1.16

msfussell committed:
Merge branch 'v1.16' into add-clickhouse-state-store-docs-v1.16

marcduiker committed:
Merge branch 'v1.16' into add-clickhouse-state-store-docs-v1.16

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker commented:
@middt I've resolved a merge conflict in genericl.yaml since the `query` property does not exist any longer. I've set the `workflow` property to false. Can you confirm if this is correct?

middt committed:
Merge branch 'v1.16' into add-clickhouse-state-store-docs-v1.16

middt commented:
> @middt I've resolved a merge conflict in genericl.yaml since the `query` property does not exist any longer. I've set the `workflow` property to false. Can you confirm if this is correct?

Yes, setting workflow: false is correct — good catch! Thanks for the fix! ✅ — @marcduiker

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,193 @@
+
+Dapr can use any ClickHouse instance: containerized, running on your local dev machine, or a managed cloud service.
+
+{{< tabs "Self-Hosted" "Kubernetes" "Cloud" >}}
```
```suggestion
{{< tabpane text=true >}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,193 @@
+
+{{< tabs "Self-Hosted" "Kubernetes" "Cloud" >}}
+
+{{% codetab %}}
```
```suggestion
{{% tab "Self-Hosted" %}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
````diff
@@ -0,0 +1,193 @@
+   CREATE DATABASE IF NOT EXISTS dapr_state;
+   ```
+
+{{% /codetab %}}
````
```suggestion
{{% /tab %}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,193 @@
+
+{{% /codetab %}}
+
+{{% codetab %}}
```
```suggestion
{{% tab "Kubernetes" %}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
````diff
@@ -0,0 +1,193 @@
+     value: "clickhouse://clickhouse:9000"
+   ```
+
+{{% /codetab %}}
````
```suggestion
{{% /tab %}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,193 @@
+
+{{% /codetab %}}
+
+{{% codetab %}}
```
```suggestion
{{% tab "Cloud" %}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,193 @@
+
+When using a managed service, ensure you have the correct connection URL, database name, and credentials configured in your component metadata.
+
+{{% /codetab %}}
```
```suggestion
{{% /tab %}}
```

marcduiker commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md:
```diff
@@ -0,0 +1,193 @@
+
+{{% /codetab %}}
+
+{{< /tabs >}}
```
```suggestion
{{< /tabpane >}}
```

marcduiker reviewed (approved):
Some syntax changes due to Hugo upgrade in 1.16

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-clickhouse.md

Signed-off-by: Marc Duiker <marcduiker@users.noreply.github.com>

marcduiker committed:
Merge branch 'v1.16' into add-clickhouse-state-store-docs-v1.16

msfussell commented:
@middt - This PR does need to target the v1.17 docs branch now. Can you update this?

middt committed:
Align ClickHouse docs with component implementation

- Rename clickhouseURL to clickhouseUrl to match component mapstructure tag
- Update URL scheme from clickhouse:// to tcp:// to match component metadata
- Update since version from 1.16 to 1.17 to match target branch
- Reword bulk operations as convenience methods (they iterate individually)

Signed-off-by: Mehmet TOSUN <mehmet.tosun@gmail.com>

middt commented:
> @marcduiker The docs PR has been retargeted to v1.17 and updated to align with the latest component implementation. The components-contrib PR is ready for re-review as well!

middt committed:
Merge branch 'v1.17' into add-clickhouse-state-store-docs-v1.16
